### DISCUSSION TITLE
Deprecation of smart banner fixes

### DISCUSSION THREAD

**Discussion by merbin2012** [admob-plus/admob-plus#688] [Show and tell]
Long day I am seeing "'SMART_BANNER: AdSize' is deprecated. Deprecated in Java"
in \src\main\kotlin\admob\plus\core\util.kt . Now I have found the solution, to change it to adaptive banner ad. Kindly check the following code and update it, if all are working fine.

36th line remove "buildAdSize" method and add the following code

    
    fun buildAdSize(opts: JSONObject, activity: Activity): AdSize {
    val name = "size"
    if (!opts.has(name)) {
        // Replace SMART_BANNER with Adaptive Anchored Banner
        val width = pxToDp(Resources.getSystem().displayMetrics.widthPixels)
        return AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(activity, width)
    }    
    val adSizeObj = opts.optJSONObject(name)
    val adSize = AdSizeType.getAdSize(opts.optInt(name))
    
    if (adSizeObj == null) {
        // Fallback to Adaptive Anchored Banner if size is not defined
        val width = pxToDp(Resources.getSystem().displayMetrics.widthPixels)
        return AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(activity, width)
    }
    
    val adaptive = adSizeObj.optString("adaptive")
    val w = pxToDp(if (adSizeObj.has("width")) adSizeObj.optInt("width") else Resources.getSystem().displayMetrics.widthPixels)
    
    if ("inline" == adaptive) {
        if (adSizeObj.has("maxHeight")) {
            return AdSize.getInlineAdaptiveBannerAdSize(w, pxToDp(adSizeObj.optInt("maxHeight")))
        }
    } else {
        return when (adSizeObj.optString("orientation")) {
            "portrait" -> AdSize.getPortraitAnchoredAdaptiveBannerAdSize(activity, w)
            "landscape" -> AdSize.getLandscapeAnchoredAdaptiveBannerAdSize(activity, w)
            else -> AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(activity, w)
        }
    }
    
    return AdSize(w, pxToDp(adSizeObj.optInt("height")))
   }

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
**SMART_BANNER** is deprecated for years, I guess even 5 years and longer.

So we don't have to worry it won't work in the short time 😃 

But you can set the `default banner` to a `adaptive banner` in the configuration. I am not sure that is the "real" adaptive banner, but the look and feel of the banner does change.

![image](https://github.com/user-attachments/assets/fed085ce-cc52-4fdf-a61d-e179f92f8304)

On the bottom you can see the `adaptive banner`, and it always fits the screen in both width and height. I guess that's why it's called "adaptive" (like responsive web-design will also auto-adjust).

`banner = new admob.BannerAd( {
  adUnitId: 'xxx-xxx-xxx',
  size: {
    adaptive: 'anchored'
  }
}  )`

The `size` parameter will show the `adaptive` banner and you can also try another format;

`size: {
  adaptive: 'inline',
  maxHeight: 150,
}`

> **Anchor adaptive banners** are always on-screen, anchored to the top or bottom of the screen with a relatively small max height. **Inline adaptive banners** are intended to be placed inside scrollable content and can be much larger as the user will typically be able to scroll past them.

Google quote from above : https://support.google.com/admob/answer/7311346

Some visual explanation about adaptive types : https://developers.google.com/admob/android/banner/inline-adaptive

---

**IMPORTANT :** I don't use the **beta** version of this plugin, but **community** : https://github.com/EYALIN/community-admob-plus
